src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -462,10 +468,23 @@ private void CheckModifiers(Location location, bool hasBody, bool isAutoProperty
             {
                 diagnostics.Add(AccessCheck.GetProtectedMemberInSealedTypeError(ContainingType), location, this);
             }
-            else if (IsStatic && IsDeclaredReadOnly && !_property.HasReadOnlyModifier)
+            else if (LocalDeclaredReadOnly)
             {
-                // The modifier '{0}' is not valid for this item
-                diagnostics.Add(ErrorCode.ERR_BadMemberFlag, location, SyntaxFacts.GetText(SyntaxKind.ReadOnlyKeyword));
+                if (_property.HasReadOnlyModifier)
+                {
+                    // Cannot specify 'readonly' modifiers on both property or indexer '{0}' and its accessors.
+                    diagnostics.Add(ErrorCode.ERR_InvalidPropertyReadOnlyMods, location, _property);
```

Comment:
>ErrorCode.ERR_InvalidPropertyReadOnlyMods [](start = 36, length = 41)

Are we testing `ERR_InvalidPropertyReadOnlyMods` or `ERR_StaticMemberCantBeReadOnly` below with indexers?

Reply:
Looks like no. Will add the coverage.

---
In reply to: [267981220](https://github.com/dotnet/roslyn/pull/34338#discussion_r267981220) [](ancestors = 267981220)

Reply:
Note that `StaticMemberCantBeReadOnly` can't be produced for indexers because indexers aren't allowed to be `static`.